src/webots/nodes/WbAbstractTransform.cpp

```diff
@@ -257,11 +259,25 @@ bool WbAbstractTransform::isTranslationFieldVisible() const {
 bool WbAbstractTransform::isRotationFieldVisible() const {
   if (!mIsRotationFieldVisibleReady) {
     mIsRotationFieldVisible = WbNodeUtilities::isVisible(mBaseNode->findField("rotation", true));
+    mCanBeRotated =
+      !WbNodeUtilities::isTemplateRegeneratorField(mBaseNode->findField("rotation", true)) && mIsRotationFieldVisible;
     mIsRotationFieldVisibleReady = true;
   }
   return mIsRotationFieldVisible;
 }
 
+bool WbAbstractTransform::canBeTranslated() const {
+  if (!mIsTranslationFieldVisibleReady)
+    isTranslationFieldVisible();
```

Comment:
This seems useless. Did you mean `return isTranslationFieldVisible();`?

Reply:
No it is not useless.
The value `mCanBeTranslated` is computed and cached in `isTranslationFieldVisible()`.
So, if the value of `mCanBeTranslated` has not been computed yet, we need to call `isTranslationFieldVisible()` to compute it.

Reply:
I see. I was fooled by those mutable fields.